File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        
#         l = min(len(word1), len(word2))
#         index = 0
#         res = []
        
#         while index < l:
#             res.append(word1[index])
#             res.append(word2[index])
#             index+=1
        
#         if index<len(word1):
#             res.append(word1[index:])
            
#         if index<len(word2):
#             res.append(word2[index:])
            
        
#         return "".join(res)
        
    
        result = ""
        for i in range(max(len(word1), len(word2))):
            
            if i < len(word1):
                result+=word1[i]
            
            if i < len(word2):
                result+=word2[i]
                
        
        return result
        
        
        

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        
        int l = Math.max(word1.length(), word2.length());
        
        StringBuilder result = new StringBuilder();
        
        for(int i=0; i< l; i++) {
            
            if(i< word1.length()) {
                result.append(word1.charAt(i));
            }
            if (i< word2.length()) {
                result.append(word2.charAt(i));
            }
        }
        
        return result.toString();
        
    }
}